app/models/concerns/foreman_bootdisk/compute_resources/proxmox.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+module ForemanBootdisk
+  module ComputeResources
+    module Proxmox
+      def capabilities
+        super + [:bootdisk]
+      end
+
+      def iso_upload(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        server.ssh_options = { password: fog_credentials[:proxmox_password] }
+        server.ssh_ip_address = server.node_id
+        server.username = client.credentials[:current_user].split('@').first
+        server.scp_upload(iso, '/var/lib/vz/template/iso/')
+        server.reload
+        storage = storages(server.node_id, 'iso')[0]
+        storage.volumes.any? { |v| v.volid.include? File.basename(iso) }
+      end
+
+      def iso_attach(iso, vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+        storage = storages(server.node_id, 'iso')[0]
+        volume = storage.volumes.detect { |v| v.volid.include? File.basename(iso) }
+        disks = server.disks.map { |disk| disk.split(":")[0] }.join(";")
+        config_hash = { ide2: "#{volume.volid},media=cdrom" }
+        server.update(config_hash)
+        server.update({ boot: "order=ide2;#{disks}" })
+        server.reboot
+      end
+
+      def iso_detach(vm_uuid)
+        server = find_vm_by_uuid(vm_uuid)
+
+        # get volid to delete iso after detaching from vm
+        volid = server.volumes.get("ide2").volid
+        config_hash = { ide2: "none,media=cdrom" }
+        server.update(config_hash)
```

Comment:
```suggestion
        server[:ide2] = "none,media=cdrom"
```

Reply:
Since it's a server object and not a hash so this assignment is not possible and it throw error `NoMethodError (undefined method `[]=' for #<Fog::Proxmox::Compute::Server:0x0000564003398f20>)`